src/main/java/shortestpath/pathfinder/VisitedTiles.java

```diff
@@ -0,0 +1,61 @@
+package shortestpath.pathfinder;
+
+import net.runelite.api.coords.WorldPoint;
+
+public class VisitedTiles {
+    private static final int TOTAL_PLANES = 4;
+    private static final int REGION_SIZE = 64;
+
+    // TODO: what is the max number of regions?
+    private static final int REGION_EXTENT_X = 192;
+    private static final int REGION_EXTENT_Y = 256;
+    private final VisitedRegion[] visitedRegions = new VisitedRegion[REGION_EXTENT_X * REGION_EXTENT_Y];
+
+    public boolean get(WorldPoint point) {
+        final int regionIndex = point.getRegionID();
+        final VisitedRegion region = visitedRegions[regionIndex];
+
+        if (region == null) {
+            return false;
+        }
+
+        return region.get(point.getRegionX(), point.getRegionY(), point.getPlane());
+    }
+    public boolean set(WorldPoint point) {
+        final int regionIndex = point.getRegionID();
```

Comment:
`point.getRegionID()` assumes that `REGION_EXTENT_X = 256`, so this needs to be a custom calculation